toonz/sources/toonz/sceneviewercontextmenu.cpp

```diff
@@ -81,8 +81,18 @@ SceneViewerContextMenu::SceneViewerContextMenu(SceneViewer *parent)
           parent->connect(action, SIGNAL(triggered()), SLOT(swapCompared()));
   }
 
+  // flip horizontally
+  action = commandManager->createAction(V_FlipX, this);
+  addAction(action);
+  ret = ret && parent->connect(action, SIGNAL(triggered()), SLOT(flipX()));
+
+  // flip vertically
+  action = commandManager->createAction(V_FlipY, this);
+  addAction(action);
+  ret = ret && parent->connect(action, SIGNAL(triggered()), SLOT(flipY()));
+
```

Comment:
I'm a bit reluctant to put many commands on the top layer of the context menu.
How about putting flip horizontally / vertically commands into "Flip View" submenu, and putting all "Reset xxx" commands into "Reset View" submenu like as follows?:
![image](https://user-images.githubusercontent.com/17974955/57439789-bab0b000-7281-11e9-8ace-cb43da4f7717.png)

Reply:
I will condense into submenus as requested.

If you are ok with the proposed changes in #2540,  I can remove the Flip Viewer commands in this context menu as they will be in a more easily accessible position.